src/pointer-analysis/value_set_analysis.h

```diff
@@ -14,32 +14,90 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #define USE_DEPRECATED_STATIC_ANALYSIS_H
 #include <analyses/static_analysis.h>
+#include <util/xml_expr.h>
+#include <util/xml.h>
 
 #include "value_set_domain.h"
 #include "value_sets.h"
+#include "value_set.h"
 
 class xmlt;
 
-class value_set_analysist:
+template<class Value_Sett>
+class value_set_analysis_baset:
   public value_setst,
-  public static_analysist<value_set_domaint>
+  public static_analysist<value_set_domaint<Value_Sett> >
 {
 public:
-  explicit value_set_analysist(const namespacet &_ns):
-    static_analysist<value_set_domaint>(_ns)
+  typedef value_set_domaint<Value_Sett> domaint;
+  typedef static_analysist<domaint> baset;
+  typedef typename baset::locationt locationt;
+
+  explicit value_set_analysis_baset(const namespacet &_ns):baset(_ns)
   {
   }
 
-  typedef static_analysist<value_set_domaint> baset;
-
   // overloading
-  virtual void initialize(const goto_programt &goto_program);
-  virtual void initialize(const goto_functionst &goto_functions);
+  void initialize(const goto_programt &goto_program)
+  {
+    baset::initialize(goto_program);
+  }
+  void initialize(const goto_functionst &goto_functions)
+  {
+    baset::initialize(goto_functions);
+  }
 
   void convert(
     const goto_programt &goto_program,
     const irep_idt &identifier,
-    xmlt &dest) const;
+    xmlt &dest) const
+  {
+    source_locationt previous_location;
+
+    forall_goto_program_instructions(i_it, goto_program)
+    {
+      const source_locationt &location=i_it->source_location;
+
+      if(location==previous_location)
+        continue;
+
+      if(location.is_nil() || location.get_file().empty())
+        continue;
+
+      // find value set
+      const value_sett &value_set=(*this)[i_it].value_set;
+
+      xmlt &i=dest.new_element("instruction");
+      i.new_element()=::xml(location);
+
+      for(value_sett::valuest::const_iterator
```

Comment:
Nit: the formatting on this for loop is a bit hard to follow. You might consider using `auto` to get each statement on one line or at least aligning each statement